### PR TITLE
Remove Insist from appendonly directory

### PR DESCRIPTION
**Pull request by soumyadeep2007**
Insist() is replaced with either Assert() or ereport(ERROR) as appropriate.

Please refer: https://groups.google.com/a/greenplum.org/forum/#!topic/gpdb-dev/BH1IXU6vnEw

### CODE REVIEW

File: src/backend/access/appendonly/appendonly_visimap_udf.c [hunk 1]
```diff
@@ -263,9 +263,9 @@ gp_aovisimap_hidden_info_internal(PG_FUNCTION_ARGS, Oid aoRelOid)
 			segno = fsinfo->segno;
 		}
 		else
-		{
-			Insist(false);
-		}
+			ereport(ERROR, (errmsg("invalid function context"), 
+						errdetail("storage must be either row" 
+							" or column oriented")));
```
Comment (danielgustafsson) [inline src/backend/access/appendonly/appendonly_visimap_udf.c]: The upstream convention is to not break error messages into multiple rows, but instead allow them to exceed the rowlength. This makes them easier on the eye in the code (subjective) but more importantly, easier to grep for should one want to figure out an error. We haven't really discussed adopting this style in Greenplum but I personally prefer it.

Comment (asimrp) [reply src/backend/access/appendonly/appendonly_visimap_udf.c]: I coaxed Soumyadeep into breaking those messages.  Makes sense to keep them unbroken, that's my preference too, from now on.  Thanks!

File: src/backend/access/appendonly/appendonlywriter.c [hunk 2]
```diff
@@ -1053,7 +1056,6 @@ SetSegnoForCompactionInsert(Relation rel,
 							   RelationGetRelationName(rel), RelationGetRelid(rel))));
 	}
 
-	Insist(usesegno != RESERVED_SEGNO);
```
Comment (dgkimura) [inline src/backend/access/appendonly/appendonlywriter.c]: What's the rational for removing rather than replacing with Assert?

Comment (asimrp) [reply src/backend/access/appendonly/appendonlywriter.c]: The `Insist` is equivalent to the `ereport(ERROR)` condition above.

**Pull request merged**